src/semantic_hashing_demo/main.py

```diff
@@ -0,0 +1,183 @@
+from typing import Dict, List
+
+import numpy as np
+import polars as pl
+from config import data_file, model, n, nbits, seed
+from numpy.random import RandomState
+from openai import OpenAI
+
+client = OpenAI()
+
+
+def get_embedding(text: str, model="text-embedding-3-small"):
+    """Get the embedding vector of a given text with default OpenAI embedding small model.
+    Small embedding model: 1536 len of float values.
+    Large embedding model: 3072 len of float values.
+
+    Args:
+        text (str): search query/reviews/comments.
+        model (str, optional): supported embedding model. Defaults to "text-embedding-3-small".
+
+    Returns:
+        CreateEmbeddingResponse: list of float values.
+    """
+    text = text.replace("\n", " ")
+    return client.embeddings.create(input=[text], model=model).data[0].embedding
+
+
+def hash_vector(v: List[np.float64], nbits: np.uint16) -> str:
+    """LSH random projection hash function with seeded hyperplane generation.
+
+    Args:
+        v (List[np.float64]): embedding vector.
+        nbits (np.uint16): no. of hyperplanes.
+        seed (str, optional): Seed for the random number generator. Defaults to 'subspace'.
+
+    Returns:
+        str: A binary string of length nbits.
+    """
+    # Create a RandomState instance with the seed
+    # rng = RandomState(seed_int)
+    rng = RandomState(seed)
+
+    # Generate hyperplanes using the seeded random number generator
+    plane_norms = rng.rand(int(nbits), len(v)) - 0.5
```

Comment:
This code should not be run every time, it is highly inefficient. Maybe turn this into a class and create the random planes on instantiation.

Reply:
Fixed. https://github.com/abhi3700/semantic-hashing-demo/commit/9d048e12840c73d309703f115955fc942dea2377